### PR TITLE
Add hide-to-tray on start option

### LINKED ISSUE #2660: Autostart possibilty with out active window
## Describe
A starting option i could launch out of the console without an active window. 
## Why

if I restart my system I want wattlebird staring without a window active on my screen to get notifications , for me my workflow is easier if i could abroad a active starting window like the element command 
_element-desktop --hidden_

### PR DESCRIPTION
**Pull request by TheFreeman193**
## Description

Implements a _hide on start_ option to minimize the window to tray on app launch.

This uses the same minimize technique as the `onClose` hook and the tray icon toggle, and occurs after the browser window load and splash screen finish.

Additional command line arguments `--hidden` and `--show` override the menu setting (for autorun/cron etc.).

The menu option exists as `Hide window on launch` under _Preferences/General/Other_.

### Notes

+ I generally work with C#/.NET/PowerShell and am unfamiliar with electron/node/ts. Please check that I haven't missed anything.
+ I've only added a language string for the menu item to the English locale, so translations will need to be made for `preferences/general/other/hideOnLaunch`.

## Related Issues

- Resolves #2660

## Appearance

Updated Preferences menu:

image

### CODE REVIEW

File: src/main/index.ts [hunk 1]
```diff
@@ -391,6 +403,15 @@ async function createWindow() {
...
     })
+
+    // Minimize to tray immediately if "hide on launch" selected
+    // or if --hidden arg is passed
+    if ((generalPreferences.other.hideOnLaunch || args.hidden) && !args.show) {
+      mainWindow.once('show', () => {
+        mainWindow!.hide()
+        mainWindow!.setSkipTaskbar(true)
+      })
+    }
```
Comment (h3poteto) [inline src/main/index.ts:414]: We can use this feature only for Windows and Linux, right?

Comment (TheFreeman193) [reply src/main/index.ts:414]: @h3poteto Correct. This feature likely wouldn't work properly on macOS due to the window handling differences. As such, I placed the additional setting in the *Other options* section which is hidden on macOS.

image

The code on [L409-414](https://github.com/h3poteto/whalebird-desktop/blob/763006f06b77b4a115548824aaba3dc5cc356110/src/main/index.ts#L409-L414) which acts on the setting is also in the conditional block excluding macOS ([L382](https://github.com/h3poteto/whalebird-desktop/blob/763006f06b77b4a115548824aaba3dc5cc356110/src/main/index.ts#L382)), so will never run on that platform, regardless of command line parameters or manual editing of the config files.

Comment (h3poteto) [reply src/main/index.ts:414]: OK. Then please pass the CI.

Comment (TheFreeman193) [reply src/main/index.ts:414]: I've changed the non-null assertions to null-conditional ones, which should now pass the eslint checks.

**Review by h3poteto — APPROVED** [review]

**Pull request merged**